components/vc/front/src/cockpitLights.c

```diff
@@ -0,0 +1,111 @@
+/**
+* @file cockpitLights.c
+* @brief Module source that manages IMD and BMS lights 
+*/
+
+/******************************************************************************
+*                             I N C L U D E S
+******************************************************************************/
+
+#include "cockpitLights.h"
+#include "CANIO_componentSpecific.h"
+#include "Module.h"
+#include "ModuleDesc.h"
+#include "string.h"
+#include "drv_outputAD.c"
+#include "app_vehicleState.h"
+#include "MessageUnpack_generated.h"
+
+/******************************************************************************
+ *                         P R I V A T E  V A R S
+ ******************************************************************************/
+struct
+{
+    IMD_LIGHT_state_E imd_state;
+    BMS_LIGHT_state_E bms_state;
+} cockpitLights_data;
+
+/******************************************************************************
+*                       P U B L I C  F U N C T I O N S
+******************************************************************************/
+
+IMD_LIGHT_state_E IMD_LIGHT_getState(void)
+{
+    return cockpitLights_data.imd_state;
+}
+
+BMS_LIGHT_state_E BMS_LIGHT_getState(void)
+{
+    return cockpitLights_data.bms_state;
+}
+
+CAN_digitalStatus_E cockpitLights_imd_getStateCAN(void)
+{
+    return (cockpitLights_data.imd_state == IMD_LIGHT_ON) ? CAN_DIGITALSTATUS_ON : CAN_DIGITALSTATUS_OFF;
+}
+
+CAN_digitalStatus_E cockpitLights_bms_getStateCAN(void)
+{
+    return (cockpitLights_data.bms_state == BMS_LIGHT_ON) ? CAN_DIGITALSTATUS_ON : CAN_DIGITALSTATUS_OFF;
+}
+
+static void cockpitLight_init(void)
+{
+    cockpitLights_data.imd_state = IMD_LIGHT_OFF,
+    drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_IMD_LIGHT_EN, DRV_IO_INACTIVE);
+
+    cockpitLights_data.bms_state = BMS_LIGHT_OFF,
+    drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_BMS_LIGHT_EN, DRV_IO_INACTIVE);
+}
+
+static void cockpitLights_periodic_10Hz(void)
+{
+    CAN_digitalStatus_E can_imd_status;
+    const bool IMD_LIGHT_valid = (CANRX_get_signal(VEH, BMSB_bmsStatusMem, &can_imd_status) == CANRX_MESSAGE_VALID);    
+
+    if (IMD_LIGHT_valid == true)
+    {
+        switch (cockpitLights_data.imd_state)
+        {
+            case IMD_LIGHT_ON:
+                drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_IMD_LIGHT_EN, DRV_IO_INACTIVE);
+                break;
+            case IMD_LIGHT_OFF:
+                drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_IMD_LIGHT_EN, DRV_IO_ACTIVE);
+                break;
+            default:
+                drv_outputAD_setDigitalActiveState(DRV_OUTPUTAD_DIGITAL_IMD_LIGHT_EN, DRV_IO_INACTIVE);
+                break;
+        }
+    }
```

Comment:
This is incorrect, you have to set the state from the CAN state, and then from that state set the light